controllers/array_action/array_mediator_svc.py

```diff
@@ -1277,6 +1365,35 @@ def get_replication(self, replication_request):
         logger.info("found rcrelationship: {}".format(rcrelationship))
         return self._generate_replication_response(rcrelationship)
 
+    def _get_ear_replication(self, replication_request):
+        if not self._is_earreplication_supported():
+            logger.info("EAR replication is not supported on the existing storage")
+            return None
+
+        # for phase 1 - find volume by id and get volume group id from result volume
+        cli_volume = self._get_cli_volume(replication_request.volume_internal_id)
+        volume_group_id = cli_volume.volume_group_id
+        if volume_group_id == "":
+            return None
+
+        cli_volume_group = self._lsvolumegroup(volume_group_id)
+        if not cli_volume_group:
+            return None
+        cli_volume_group_replication = self._lsvolumegroupreplication(volume_group_id)
+        replication_mode = self._get_replication_mode(volume_group_id)
+        logger.info("found ear replication: {} in mode: {}".format(cli_volume_group_replication.replication_policy_name,
+                                                                   replication_mode))
+        return self._generate_ear_replication_response(cli_volume_group_replication, replication_mode)
```

Comment:
I think it can be exported to `_generate_ear_response_from_volume_group` function

Reply:
Done